gazebo_grasp_plugin/src/GazeboGraspFix.cpp

```diff
@@ -559,43 +562,71 @@ void GazeboGraspFix::OnUpdate() {
+
+            // Get transform for currLinkWorldPose as matrix
+            gz_math_Matrix4d worldToLink(currLinkWorldPose.Rot());
+            worldToLink.SetTranslation(currLinkWorldPose.Pos());
+
+            // Get the transform from collision link to contact point
+            gz_math_Matrix4d linkToContact=gz_math_Matrix4d::Identity;
+            linkToContact.SetTranslation(relContactPos);
+
+            // the current world position of the contact point right now is:
+            gz_math_Matrix4d _currContactWorldPose=worldToLink*linkToContact;
+            gz_math_Vector3d currContactWorldPose=_currContactWorldPose.Translation();
+#else
+            gz_math_Pose3d currObjWorldPose=cpInfo.collObj->GetLink()->GetWorldPose();
 
             // get world pose of link
-            gazebo::math::Pose currLinkWorldPose=cpInfo.collLink->GetLink()->GetWorldPose();
+            gz_math_Pose3d currLinkWorldPose=cpInfo.collLink->GetLink()->GetWorldPose();
 
             // Get transform for currLinkWorldPose as matrix
-            gazebo::math::Matrix4 worldToLink=currLinkWorldPose.rot.GetAsMatrix4();
+            gz_math_Matrix4d worldToLink=currLinkWorldPose.rot.GetAsMatrix4();
             worldToLink.SetTranslate(currLinkWorldPose.pos);
 
             // Get the transform from collision link to contact point
-            gazebo::math::Matrix4 linkToContact=gazebo::math::Matrix4::IDENTITY;
+            gz_math_Matrix4d linkToContact=gz_math_Matrix4d::IDENTITY;
             linkToContact.SetTranslate(relContactPos);
-                    
+
             // the current world position of the contact point right now is:
-            gazebo::math::Matrix4 _currContactWorldPose=worldToLink*linkToContact;
-            gazebo::math::Vector3 currContactWorldPose=_currContactWorldPose.GetTranslation();
+            gz_math_Matrix4d _currContactWorldPose=worldToLink*linkToContact;
+            gz_math_Vector3d currContactWorldPose=_currContactWorldPose.GetTranslation();
+#endif
 
             // the initial contact point location on the link should still correspond
             // to the initial contact point location on the object.
 
             // initial vector from object center to contact point (relative to link,
             // because relObjPos and relContactPos are from center of link)
-            gazebo::math::Vector3 oldObjDist= relContactPos - relObjPos;
+            gz_math_Vector3d oldObjDist= relContactPos - relObjPos;
             // the same vector as \e oldObjDist, but calculated by the current world pose
             // of object and the current location of the initial contact location on the link.
-            gazebo::math::Vector3 newObjDist= currContactWorldPose - currObjWorldPose.pos; // new distance from contact to object
+#if GAZEBO_MAJOR_VERSION >= 8
+            gz_math_Vector3d newObjDist= currContactWorldPose - currObjWorldPose.Pos(); // new distance from contact to object
```

Comment:
See my comment above with the getters. It applies to all such cases, so I won't put a comment everywhere ;)